### ISSUE TITLE
Add heading numbering to Antora output

### ISSUE THREAD

**Issue by nakohdo** [asam-ev/OpenMATERIAL#66]
## Problem
The [Antora output of OpenMATERIAL](https://asam-ev.github.io/OpenMATERIAL) does not contain numbered headings as other ASAM documentation, for example, [ASAM OSI](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/latest/specification/index.html). This problem has been reported in #54. 

## Cause
The current, basic Antora configuration does not contain the setting for heading numbering yet. These settings are part of the [ASAM Antora Extensions](https://common.pages.asam.net/antora-ui/ASAM_Antora_Extensions/antora/consistent_numbering/description.html).

## Solution 
Add the configuration to the Antora playbook file. 

```YAML
antora:
  extensions:
  - require: './asam-antora-extensions/asam-antora_extension.js'
    numbered_titles: true
    section_number_style: "iso"
```

**Comment by nakohdo** [comment]
The ASAM Antora Extensions contain lots of internal dependencies so just adding the above code to the playbook file unfortunately does not work.

**Comment by nakohdo** [comment]
I have successfully implemented the full Antora Generator pipeline in my testing environment. Must still be transferred to the production environment.

https://nakohdo.github.io/openmaterial-antora-generator/asamopenmaterial/latest/specification/index.html

**Comment by ClemensLinnhoff** [comment]
Looks very good.
Just one question: How is the split between left and right menu defined?
In you example, you have all first level chapters and all sub-chapters on the right, respectively.
In OpenDRIVE and OpenSCENARIO, also the seconds level chapters are in the left menu.
I actually like having the second level chapters, e.g. 6.2) also on the left, but this is just personal preference. You would get one more level in the menu on the right side this way. Is there an ASAM convention for this?

**Comment by nakohdo** [comment]
@ClemensLinnhoff 
I don't think there is an ASAM convention for this because the numbering looks slightly different between projects. We took the basic configuration from OSI but OSI looks different than OpenMATERIAL at the moment. We are still trying to figure out why that's the case.

**Comment by ClemensLinnhoff** [comment]
I think using OSI as a reference is probably not good, because OSI has the special case, that the documentation is constructed from different repositories. In OpenMATERIAL we just have one.

**Comment by nakohdo** [comment]
We are currently looking into this.

**Comment by martinfiebig** [comment]
Hi Clemens,
we have now extensively tested the asam-antora-extension for the numbered headings and found the following options:

1. Option is to add xref elements to the subchapters in the index.adoc file. Also alter the leveloffset-values in the subchapters. This would produce the follwing output (example chapter 7 Geometry):
![Bild (1)](https://github.com/user-attachments/assets/58ada97b-a67b-43b0-909a-fa0514e6b303)
The disadvantage here is that the TOC on the right side would dissapear when you click on the Geometry chapter:
![Bild (2)](https://github.com/user-attachments/assets/cf638738-d022-483d-b3f0-e22f470471f4)

2. Option is to create a title-topic for the chapters 'Geometry' and 'Material' and also change the structure of the index.adoc and the sub-chapter files. This would create the following output:
![Bild (3)](https://github.com/user-attachments/assets/ea1d61c3-3582-4195-a16c-de37f64361e3)
The disadvantages here are that when you click on the 'Geometry' chapter, you'll only see the title-topic:
![Bild (4)](https://github.com/user-attachments/assets/33e1d952-6d05-49b9-9722-d141a7d1e277)
This is the solution, that seems to be implemented for OSI (i.e. [here](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/latest/interface/open-simulation-interface_user_guide.html)) and OpenSCENARIO (i.e. [here](https://www.asam.net/static_downloads/public/asam-openscenario/2.0.0/language-reference/language-reference.html)) - although OpenSCENARIO has added links to their respective subchapters, which would be possible for OpenMATERIAL as well.
This solution however has the highest maintenance effort, because for i.e. the addition of new topics, multiple files would have to be changed/updated.

3. Option is to create a separate nav.adoc file, in which the navigation on the left side is defined. The attribute 'antora_mapping' would then be set in this file rather than in the index.adoc file. This would produce the following output:
![image](https://github.com/user-attachments/assets/73c79873-01e9-49f2-8767-c1ae9718b0a8)
This would be the cleanest solution, as the navigation is independent from the structure of the index.adoc file. To add new chapters or topics to the documentation, you would insert them to the nav.adoc file - and only change the index.adoc, if the newly added content should be shown there as well.

We would recommend to use the 1. or 3. option, as the 2. option would, as stated above, noticeably increase the effort for the maintenance of the documentation. It also makes the process of adding new bits to the documentation more complex.

Please decide and let us know which of the above solutions would work best and should therefore be implemented.

**Comment by ClemensLinnhoff** [comment]
I am not entirely sure I understand option 3. Isn't that what we have currently? We have this navigation file, here:
https://github.com/asam-ev/OpenMATERIAL/blob/main/.antora/modules/specification/nav_openmaterial.adoc

**Comment by martinfiebig** [comment]
Yes, for the current workflow, that is correct. However, to add the functionality for the numbering, we had to include the asam-antora-extensions (see [here](https://code.asam.net/common/asam-antora-extensions)). By default, these extensions also generate a nav-file, based on the index.adoc - which unfortunately overwrites the nav_openmaterial.adoc. So yes, for option 3, we could take the existing nav-file, move it to the contents/ folder and set the required attributes for the asam-antora-extensions.

**Comment by ClemensLinnhoff** [comment]
Okay sound good. Then I would prefer that option. In this option the navigation on the right would stay as it is now? So only showing the content of the selected chapter? Because I actually like this a lot and I think this is also the way it is in OpenSCENARIO.

**Comment by ClemensLinnhoff** [comment]
This was fixed with #114. Thanks @martinfiebig!
However, I have some questions about the solution:

1. The jobs "Checkout documentation" and "Apply dispatch information" are not executed. The generated documentation page seems okay, though. So are there jobs needed? See https://github.com/asam-ev/OpenMATERIAL/actions/runs/11051822058/job/30702552731 
2. material/fileformat.adoc is not found. This is probably just misspelled and it should be material/file-format.adoc. See error message here: https://github.com/asam-ev/OpenMATERIAL/actions/runs/11051822058/job/30702552731#step:9:418 
3. There are some warnings about unlisted pages: https://github.com/asam-ev/OpenMATERIAL/actions/runs/11051822058/job/30702552731#step:9:418
Could you please check where these warnings come from?

**Comment by ClemensLinnhoff** [comment]
I just found another issue:
4. The number of the material properties is missing: 
![image](https://github.com/user-attachments/assets/f7a3c899-f3df-4043-b2ab-1c9b4b698d7a)
This is probably because I manually changed the name in the nav file. It is probably a better idea to change this in the asciidoc generation from the schema files. Then we don't need the manual naming in the nav anymore. I will create a separate issue for this and take a look at it myself.

**Comment by martinfiebig** [comment]
Hi @ClemensLinnhoff, thanks for merging the pull request!

Regarding 1): These steps are inherited from the example configuration of the asam-antora-extensions. They only seem to be executed, when the build-workflow is triggered by the workflow-trigger 'repository-dispatch', so if i.e. something gets published in a different repository. As we do not have this case for OpenMATERIAL, we can remove these steps. I will create a pull request for this.

Regarding 2): Yes, this is actually a typo. I will also fix that in above pull request.

Regarding 3): These warnings come from one of the asam-antora-extensions: [Orphan pages and partials](https://code.asam.net/common/asam-antora-extensions#user-content-orphan-pages-and-partials). This extension finds and lists all pages, that are not referenced in the nav-file of the project. So there is no actual problem with the files, they are just not referenced in the nav-file. I'd say we can safely ignore these warnings.

**Comment by ClemensLinnhoff** [comment]
Thanks for 1 and 2.

For 3: I am not a big fan of ignoring warnings and I'd rather adopt a treat-warnings-as-errors policy. Because otherwise valid warnings might get missed. Could you check if there is any possibility to safely deactivate this warning? If there is none, we should create an option in the asam-antora-extensions repo. Or can we change something in our file structure to avoid this warning? We'd probably have to put everything beneath the second headline level in one file, right? I don't think this is really an option.

**Comment by martinfiebig** [comment]
Hi Clemens,
I've create the pull request #124 for 1 and 2.

For 3: I agree, the warnings shouldn't appear in the first place. I will investigate, if and how they can be turned off. I've created issue #123 for that, so we could close this issue already.

**Comment by ClemensLinnhoff** [comment]
Thanks, I have merged #124.
